File: sandbox-go/path.go
package sandbox

import (
	"net/url"
	"strings"
)

func encodeFilePath(path string) string {
	segments := strings.Split(path, "/")
	encoded := make([]string, 0, len(segments))
	for _, segment := range segments {
		if segment == "" {
			continue
		}
		encoded = append(encoded, url.PathEscape(segment))
	}
	return strings.Join(encoded, "/")
}

File: sandbox/src/resources/path.ts
/** Encode a single path segment safely for URL usage. */
export function encodePathSegment(value: string): string {
  return encodeURIComponent(value);
}

/**
 * Encode a sandbox file path without encoding forward slashes.
 * Example: `tmp/my file.txt` -> `tmp/my%20file.txt`
 */
export function encodeFilePath(path: string): string {
  return path
    .split('/')
    .filter((segment) => segment.length > 0)
    .map(encodePathSegment)
    .join('/');
}
